controllers/build_controller.go

```diff
@@ -0,0 +1,315 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"strings"
+)
+
+// BuildReconciler reconciles a Build Pod object
+type BuildReconciler struct {
+	client.Client
+	Scheme            *runtime.Scheme
+	InsightsJWTSecret string
+	ScanImageName     string
+}
+
+const insightsScannedLabel = "insights.lagoon.sh/scanned"
+const dockerhost = "docker-host.lagoon.svc" //TODO in future versions this will be read from the build CRD
+
+//+kubebuilder:rbac:groups=core,resources=namespaces,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=core,resources=namespaces/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=core,resources=namespaces/finalizers,verbs=update
```

Comment:
Is this used for anything? I don't see any finalizer interactions?

Reply:
nope, I think that's just boilerplate - I'll remove it.

Reply:
usually the `kubebuilder:rbac` stuff is used by things built by kubebuilder to template out `Role` and `ClusterRole` for the service account. Would be beneficial to leave whatever is sufficient at least for the operation of this bit of code